system-addon/content-src/components/TopSites/_TopSites.scss

```diff
@@ -337,20 +337,70 @@ $half-base-gutter: $base-gutter / 2;
         transform: translateY(-50%);
         top: 50%;
         offset-inline-end: 8px;
+      }
+    }
 
-        & + input:dir(ltr) {
-          padding-right: 32px;
-        }
+    .url {
+      input:dir(ltr) {
+        padding-right: 32px;
+      }
 
-        & + input:dir(rtl) {
-          padding-left: 32px;
+      input:dir(rtl) {
+        padding-left: 32px;
+
+        &:not(:placeholder-shown) {
+          direction: ltr;
+          text-align: right;
         }
       }
     }
 
-    .url input:not(:placeholder-shown):dir(rtl) {
-      direction: ltr;
-      text-align: right;
+    .enable-custom-image-input {
+      display: inline-block;
+      font-size: 13px;
+      margin-top: 4px;
+      cursor: pointer;
+
+      &:hover {
+        text-decoration: underline;
+        color: $blue-60;
+      }
+    }
+
+    .custom-image-input-container {
+      margin-top: 4px;
+
+      .loading-container {
+        width: 16px;
+        height: 16px;
+        overflow: hidden;
+        position: absolute;
+        transform: translateY(-50%);
+        top: 50%;
+        offset-inline-end: 8px;
+      }
+
+      // See https://searchfox.org/mozilla-central/rev/3abf6fa7e2a6d9a7bfb88796141b0f012e68c2db/browser/themes/shared/tabs.inc.css#191
```

Comment:
This comment still isn't useful. It still requires someone to check the reference to figure out what it's even talking about. Maybe something like "This animation is derived from Firefox's tab loading animation: …"